inspire/dojson/hep/fields/bd01x09x.py

```diff
@@ -95,15 +95,23 @@ def system_control_number2marc(self, key, value):
 
 
 @hep.over('report_number', '^037..')
-@utils.for_each_value
-@utils.filter_values
+#@utils.for_each_value
+#@utils.filter_values
 def report_number(self, key, value):
     """Source of Acquisition."""
-    return {
-        'primary': value.get('a', value.get('z')),
-        'arxiv_category': value.get('c'),
-        'source': value.get('9')
-    }
+    def get_value(value):
+        return {
+            'primary': bool(value.get('a')),
```

Comment:
Uh! Actually primary is whether the reportnumber is primary (hence `037`) or not (hence `088` tag). $a contains instead what should into the `value` JSON key. So in the context of `037` you can always put `True` in the `primary` field.